### ISSUE TITLE
Proposal: calculate everything in days for stock, then convert to months, weeks, etc on the client

### ISSUE THREAD

**Issue by jniles** [third-culture-software/bhima#5008]
We recently had some confusion over units (https://github.com/IMA-WorldHealth/bhima/pull/5004) where the lifetime was calculated in months and the validation code expected it to be in days.  It's really important for us to get units right.

For Health Centers, we already know that we'll need to calculate statistics using days of stock out - I don't think we will ever need a smaller unit than this.  Therefore, I propose that **all stock units sent from the server be in days**.  If we need the values in weeks, or months, or trimesters, or years, it's very easy to convert simply by multiplying.  The only exception to this will be the **CMM**, since it is named and has a definition of we all know.

Therefore, instead of "months of stock" sent on the lots, we will have "days of stock".   For expiration, we'll send the *days until the stock expires*.  For stock outs, we'll send the *days until stock out*.  And so on.

@jmcameron @lomamech @mbayopanda What do you think?

**Comment by mbayopanda** [comment]
What are the units here?

before the introduction of `stock_movement_status`, we were storing consumption by period (month) which help us to calculate CMM, if units are (stock months, lot lifespan, risky lots) these units were based on CMM which was based on periods (months).

With the introduction of `stock_movement_status`, are these units supposed to be defined by days from the `stock_movement_status` or what ?

**Comment by jniles** [comment]
Yes, the proposal is to use days as the base unit for everything.  I suppose if we decided, we could eliminate CMM as well from the API and simply calculate it on the client as well.

So, the server API should return:

- Days of stock remaining - not number of months of stock remaining.
- Days of stock until reorder is necessary - not number of months until reorder.
- Days until the stock expires - not months until the stock expires.
